### PR TITLE
Identity contract (part 2)

### PR DESCRIPTION
**Pull request by moskvanaft**
More tests, bugfixes, etc

### CODE REVIEW

File: contracts/eosiolib/singleton.hpp [hunk 1]
```diff
@@ -27,12 +27,18 @@ namespace  eosio {
             char temp[1024+8];
             *reinterpret_cast<uint64_t *>(temp) = SingletonName;
             auto read = load_i64( scope, Code, SingletonName, temp, sizeof(temp) );
-            assert( read > 0, "singleton does not exist" );
-            datastream<const char*> ds(temp + sizeof(SingletonName), read);
+            assert( read < 0, "singleton does not exist" );
```
Comment (heifner) [inline contracts/eosiolib/singleton.hpp]: The assert was correct before.

Comment (moskvanaft) [reply contracts/eosiolib/singleton.hpp]: fixed. Thanks!

File: libraries/testing/include/eosio/testing/tester.hpp [hunk 2]
```diff
@@ -14,6 +15,8 @@ namespace eosio { namespace testing {
     */
    class tester {
       public:
+         typedef string ActionResult;
```
Comment (heifner) [inline libraries/testing/include/eosio/testing/tester.hpp]: Do not use camel case for types.

Comment (moskvanaft) [reply libraries/testing/include/eosio/testing/tester.hpp]: fixed

**Pull request merged**